src/autobot/iautobot.h

```diff
@@ -28,8 +34,22 @@ class IAutobot : MODULE_EXPORT_INTERFACE
 public:
     virtual ~IAutobot() = default;
 
-    virtual void run() = 0;
+    enum class Status {
+        Stoped = 0,
+        RunningAll,
+        RunningFile
+    };
+
+    virtual std::vector<ITestCasePtr> testCases() const = 0;
+    virtual ITestCasePtr testCase(const std::string& name) const = 0;
+
+    virtual void runAll(const std::string& testCaseName) = 0;
+    virtual void runFile(const std::string& testCaseName, int fileIndex) = 0;
```

Comment:
Based on the name of the method, I'd expect to have only one argument like filePath

`runFile(const std::string& filePath)`

could you explain the meaning of the second argument?

Reply:
first arg - test case, which needs run
second arg - file as file index in the file list (just for optimization, avoiding search file by path and etc)